src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -198,6 +198,11 @@ public Stream<SourceFile> listSourceFiles(MavenProject mavenProject, @Nullable X
             return sourceFile;
         }).filter(Objects::nonNull);
 
+        // Clear the charset after parsing or sources is complete, as it may not be applicable to other sources.
```

Comment:
I am not convinced that this actually works. Since we are returning a stream of source files, the parsing doesn't actually occur until after the return from this method.

Reply:
Or is the value from the execution context already read before any parsing starts?